internal/workflow.go

```diff
@@ -73,10 +73,22 @@ type (
 	// Selector must be used instead of native go select by workflow code.
 	// Use workflow.NewSelector(ctx) method to create a Selector instance.
 	Selector interface {
-		AddReceive(c Channel, f func(c Channel, more bool)) Selector
-		AddSend(c Channel, v interface{}, f func()) Selector
-		AddFuture(future Future, f func(f Future)) Selector
-		AddDefault(f func())
+		AddReceive(c Channel, handlerFunc func(c Channel, more bool)) Selector
+		AddSend(c Channel, v interface{}, handlerFunc func()) Selector
+		// AddFuture adds a future to the selector,
+		// calling Get on the provided future inside the handlerFunc is guaranteed not to block,
+		// hence IsReady will return true.
+		// If AddFuture is called after Select, then ???.
```

Comment:
If AddFuture is called after Select, the selector will not execute the handler function.
If AddFuture is called with the same future twice, the selector will be executed twice